### ISSUE TITLE
Signals are lost in some cases

### ISSUE THREAD

**Issue by mehtonen-boox** [kefirjs/kefir#271]
I'm not sure if this is a bug, but I found this strange behavior of signals getting lost on some cases.

```
const pool = kefir.pool();
const prop = pool.toProperty(() => 0);
const later = kefir.later();

const stream = later
 .flatMapLatest(() => kefir.merge([prop.take(1), pool]))
 .log('stream');

pool.plug(stream.take(1).map(() => 1));
```

I would expect **stream** to log with 0 and 1, but only 0 gets logged. Or I'm I misunderstanding this somehow?

**Comment by ivan-kleshnin** [comment]
1) `K.later` should be called like `K.later(time, value)`. Docs don't mention any defaults for this function.

2) https://kefirjs.github.io/kefir/#later
> laterKefir.later(wait, value)

Creates a stream that **produces a single value after wait milliseconds, then ends**.

**Comment by mehtonen-boox** [comment]
@ivan-kleshnin issue was originally produced without using **later**. It was the most simplest way to make stream give signal async. I updated description and added something to highlight the problem a little bit more.

**Comment by ivan-kleshnin** [comment]
I would say then, that the strange case is:

```
const stream = later
 .flatMapLatest(() => prop)
 .log('stream');
```

producing two values. Because `later` has to end after the first event. And your `stream` is made `later` combinator so it should end as soon as `later` ends.

**Comment by mehtonen-boox** [comment]
@ivan-kleshnin in your case when flatMapLatest is run, all the constant signals are already sent. You will get only the latest value by using `prop.take(1)`, which is 3. **Pool** won't get any new signals and therefor only **3** is logged. In this case this works as expected.

In my case **pool** should give signal with value of **1**, because that signal is sent after flatMapLatest.

**Comment by ivan-kleshnin** [comment]
Yep, makes sense. I never used `plug` for async stuff so I didn't get what you're trying to demonstrate here.

**Comment by tweinfeld** [comment]
@mehtonen-boox I think this is actually the expected result:
`pool` becomes active a split second after the `prop` constant is consumed by `stream`, at which point it will never receive any values from it via `plug`. That's because `.merge()` processes streams in order.
Evidently, if you switch the order in which you `merge` the streams, so that instead of:
```js
Kefir.merge([prop.take(1), pool]);
```
You'd use
```js
Kefir.merge([pool, prop.take(1)]);
```

`stream` would produce the expected result!

**Comment by mehtonen-boox** [comment]
@tweinfeld Thank you! I didn't know the order matters while merging streams. 🤔 Now when I know this I might be able to avoid issues like this. 👍

**Comment by ivan-kleshnin** [comment]
@mehtonen-boox it matters and there's also a difference between static functions and methods.

For example `K.merge([x1, x2])` may work different than `x1.merge(x2)` especially with cyclic dependencies.

**Comment by rpominov** [comment]
> there's also a difference between static functions and methods

This should be false, at least my intent was to make them exactly the same. For example `x1.merge(x2)` basically does `K.merge([x1, x2])` internally:

https://github.com/kefirjs/kefir/blob/bacac22c4dc6d80ef68acf933dd08a450d80692c/src/index.js#L305-L307

**Comment by ivan-kleshnin** [comment]
Sorry for bad description. What I wanted to say is that sometimes:

```js
x.skip(1).merge(y)
```

won't work (and `y.merge(x.skip(1))` is a different thing) so your only option may be a switch to statics:

```js
K.merge([
  y,
  x.skip(1),
])
```
